orangecontrib/network/widgets/OWNxGroups.py

```diff
@@ -0,0 +1,204 @@
+import numpy as np
+
+from Orange.data import DiscreteVariable, Table, Domain
+from Orange.widgets import gui
+from Orange.widgets.settings import ContextSetting, DomainContextHandler
+from Orange.widgets.utils.itemmodels import DomainModel
+from Orange.widgets.widget import Input, Output, OWWidget, Msg
+from orangecontrib.network import Graph
+
+
+class OWNxGroups(OWWidget):
+    name = "Network Of Groups"
+    description = "Group instances by feature and connect related groups."
+    icon = "icons/NetworkGroups.svg"
+    priority = 6435
+
+    class Inputs:
+        network = Input("Network", Graph, default=True)
+        data = Input("Data", Table)
+
+    class Outputs:
+        network = Output("Network", Graph, default=True)
+        data = Output("Data", Table)
+
+    class Warning(OWWidget.Warning):
+        no_graph_found = Msg("Data is given, network is missing.")
+        no_discrete_features = Msg("Data has no discrete features.")
+
+    class Error(OWWidget.Error):
+        data_size_mismatch = Msg("Length of the data does not "
+                                 "match the number of nodes.")
+
+    resizing_enabled = False
+    want_main_area = False
+
+    settingsHandler = DomainContextHandler()
+    feature = ContextSetting(None)
+
+    def __init__(self):
+        super().__init__()
+        self.network = None
+        self.data = None
+        self.effective_data = None
+        self.output_network = None
+
+        info_box = gui.widgetBox(self.controlArea, "Info")
+        self.input_label = gui.widgetLabel(info_box, "")
+        self.__set_input_label_text()
+
+        feature_box = gui.widgetBox(self.controlArea, "Group by")
+        self.feature_model = DomainModel(valid_types=DiscreteVariable)
+        self.feature_combo = gui.comboBox(
+            feature_box, self, "feature", contentsLength=15,
+            callback=self.__feature_combo_changed, model=self.feature_model
+        )
+
+    def __set_input_label_text(self):
+        text = "No data on input."
+        if self.network is not None:
+            dir_ = "Directed" if self.network.is_directed() else "Undirected"
+            text = f"{dir_} graph\n{self.network.number_of_nodes()}" \
+                   f" nodes, {self.network.number_of_edges()} edges"
+        self.input_label.setText(text)
+
+    def __feature_combo_changed(self):
+        self.commit()
+
+    @Inputs.network
+    def set_network(self, network):
+        self.network = network
+        self.__set_input_label_text()
+
+    @Inputs.data
+    def set_data(self, data):
+        self.data = data
+
+    def handleNewSignals(self):
+        if self.feature_model:
+            self.closeContext()
```

Comment:
This `if` is unnecessary; `closeContext` contains a check whether a context is currently open.